components/public-api-server/pkg/apiv1/prebuild.go

```diff
@@ -6,23 +6,24 @@ package apiv1
 
 import (
 	"context"
+
+	"github.com/bufbuild/connect-go"
 	v1 "github.com/gitpod-io/gitpod/public-api/v1"
+	"github.com/gitpod-io/gitpod/public-api/v1/v1connect"
 )
 
 func NewPrebuildService() *PrebuildService {
-	return &PrebuildService{
-		UnimplementedPrebuildsServiceServer: &v1.UnimplementedPrebuildsServiceServer{},
-	}
+	return &PrebuildService{}
 }
 
 type PrebuildService struct {
-	*v1.UnimplementedPrebuildsServiceServer
+	v1connect.UnimplementedPrebuildsServiceHandler
```

Comment:
Asking for posterity: Is there an alternative to riddling our implementation with references to `connect`? It's not a big concern, as the changes look rather mechanical (near search-and-replace). Still wondering if there is a way to have a `ConnectProxy` that does the wrapping for us (I'm aware that this might be a tough ask for Go :upside_down_face: ).

Reply:
In general, we don't want to make this magic. We also want to stay as close to the official connect documentation as possible such that we don't need to re-teach things internally.

If it's just a wrapper, it's not adding match aside from importing the function from elsewhere (and that place imports connect).

Reply:
> We also want to stay as close to the official connect documentation as possible

IMO the ideal situation would be if we could stick to the gRPC documentation. But I can definitely live with this. :100: It's not in the interest of connect, but maybe someone writes a wrapper at some point that we can use. :slightly_smiling_face:

Reply:
Yes, but if we stuck with gRPC documentation then you'd not know you can invoke it with curl and use JSON as the payload - in Go principles we should not hide things, and the difference (or complexity) should be visible directly

Reply:
> in Go principles

I know, I know... but abstraction is a powerful concept (if it does not leak too much, hence I'm fine with as-is).